cleancat/base.py

```diff
@@ -160,7 +160,9 @@ def __init__(self, regex=None, regex_flags=None, regex_message=None,
             self.regex_message = regex_message
 
     def get_regex(self):
-        return re.compile(self.regex, self.regex_flags)
+        if not getattr(self, '_compiled_regex', None):
```

Comment:
`hasattr`.

Also why not compile it eagerly right in the constructor? You'll save one more check.

Reply:
`getattr` was probably used due to the issues with `hasattr` as described in this [post](https://hynek.me/articles/hasattr/)